presto-spi/src/main/java/com/facebook/presto/spi/block/AbstractMapBlock.java

```diff
@@ -160,6 +161,30 @@ public long getRegionSizeInBytes(int position, int length)
                 Integer.BYTES * HASH_MULTIPLIER * (long) entryCount;
     }
 
+    @Override
+    public long getPositionsSizeInBytes(boolean[] positions)
+    {
+        int positionCount = getPositionCount();
+        checkValidPositions(positions, positionCount);
+        boolean[] entryPositions = new boolean[getRawKeyBlock().getPositionCount()];
```

Comment:
There is an assumption here that creating the entryPositions array is cheaper than calling, `getRegionSizeInBytes` for each used position.  I'd  add a comment here to that effect.

Reply:
This is the one reason I might want to keep `getRegionSizeInBytes`. There are situations for one to be cheaper than the other. But I'm not sure to what extend we could expect people to be able to choose the right one. If we'd keep both comment would definitely help. But if we were to remove `getRegionSizeInBytes` then does that still matter?

Reply:
Should that be a separate commit or it could all be bundled in the same one?

Reply:
I do not feel particularly comfortable doing this. `getRegionSizeInBytes` is sufficient in all cases but dictionary. And it's more efficient whenever a single invocation of it is sufficient. I believe the risk of performance regression exists. For that, I prefer that the removal of getRegionSizeInBytes be done in a separate commit and PR.

Reply:
@haozhun Do you mean that we should still experiment with removing `getRegionSizeInBytes` but it's better to do it with a separate PR so if we see regression we can quickly fix it? Is there anything we can do to test whether this is likely to cause regressions?

Reply:
I don't advocate for removing getRegionSizeInBytes.